chall04/gbudau.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import sys
+
+STDIN = 0
+FILE = 1
+
+def read_error(filename):
+    print("{}: {}: Can't read file".format(sys.argv[0], filename))
+
+def space_error(filename):
+    print("{}: {}: Not enough space in the given shelves".format(sys.argv[0], filename))
+
+def fill_bookshelves(filename, bookshelves, books):
+    # Convert books and bookshelves to integers
+    # Sort from max to min
+    books = [int(i) for i in books]
+    bookshelves = [int(i) for i in bookshelves]
+    books.sort(reverse = True)
+    bookshelves.sort(reverse = True)
```

Comment:
You can use the builtin `sorted` to do improve the readability:
```suggestion
    books = sorted([int(i) for i in books], reverse=True)
    bookshelves = sorted([int(i) for i in bookshelves], reverse=True)
```

Reply:
This looks so much better, thanks for the example.
I was thinking about a way to sort and convert at the same time but didn't figure out how to do it.